dowhy/causal_discovery.py

```diff
@@ -0,0 +1,32 @@
+class GraphLearner:
+	"""Base class for an causal discovery methods.
+
+	Subclasses implement different discovery methods. All discovery methods are in the package "dowhy.causal_discoverers"
+
+	"""
+
+	def __init__(self, data, library_class, *args, **kwargs):
+
+		self._data = data
+		self._labels = list(self._data.columns)
+		self._adj_matrix = None
+		self._graph_dot = None
+		
+	def learn_graph(self):
+		'''
+		Discover causal graph and the graph in DOT format.
+
+		'''
+		raise NotImplementedError
+
+	def _get_adjacency_matrix(self):
```

Comment:
where is the get_adjacency_matrix method used?

Reply:
Just kept it in case the user wants the adjacency matrix. I thought of keeping it as an utility. Should I be doing that?

Reply:
Let's not keep it unless there is a specific usecase. Advanced users can always do graph_learner._adjacency_matrix.